baseplate/frameworks/queue_consumer/kombu.py

```diff
@@ -14,17 +16,48 @@
 from gevent.server import StreamServer
 from kombu.mixins import ConsumerMixin
 from kombu.transport.virtual import Channel
+from prometheus_client import Counter
+from prometheus_client import Gauge
+from prometheus_client import Histogram
 
 from baseplate import Baseplate
 from baseplate import RequestContext
 from baseplate.clients.kombu import KombuSerializer
+from baseplate.lib.prometheus_metrics import default_latency_buckets
 from baseplate.server.queue_consumer import HealthcheckCallback
 from baseplate.server.queue_consumer import make_simple_healthchecker
 from baseplate.server.queue_consumer import MessageHandler
 from baseplate.server.queue_consumer import PumpWorker
 from baseplate.server.queue_consumer import QueueConsumerFactory
 
 
+class AmqpConsumerPrometheusLabels(NamedTuple):
+    amqp_address: str
```

Comment:
```
* `amqp_address` - the name of the AMQP queue host with the port, e.g. `rabbitmq:5672`
* `amqp_exchange_name` - the name of the exchange being accessed, eg. `notifications`
* `amqp_exchange_type` - the name of the exchange type being used to route it from the exchange to the queue, eg `direct`, `fanout`, `topic`, or `headers`
* `amqp_queue` - the name of the queue being accessed, eg. `push`
* `amqp_success` - MUST be `false` if an exception was raised OR if an error was returned, or `true` otherwise
```

These were the labels in the spec, should the spec be updated?

Should `amqp_exchange` be `amqp_exchange_name` to match the producer spec?

Reply:
Updated to match the spwc (amqp_exchange_name) the type cannot currently be exposed in the queue consumer.

Reply:
Updated spec: https://github.snooguts.net/reddit/baseplate.spec/pull/51